### ISSUE TITLE
Always add `registry: jspm` by default to overrides with dependencies

### ISSUE THREAD

**Issue by trusktr** [jspm/jspm-cli#360]
I'm running

```
jspm install -y famous=github:Famous/famous -o '{ "directories": { "lib": "src" }, "dependencies": { "css": "^0.1.0" }, "shim": { "*/*": { "deps": ["../core/famous.css!"] } } }'
```

then getting this error in-browser:

```
Potentially unhandled rejection [5] Error loading "github:Famous/famous@0.3.4/core/famous.css!css" at <unknown> es6-module-loader.src.js:138
Error loading "github:Famous/famous@0.3.4/core/famous.css!css" from "app" at http://localhost:3000/app.js
Error loading "css" at http://localhost:3000/css.js
Not Found: http://localhost:3000/css.js (WARNING: non-Error used)
    (anonymous function) es6-module-loader.src.js:138
    report es6-module-loader.src.js:165
    flush es6-module-loader.src.js:187
```

**Comment by guybedford** [comment]
You need a `registry: "jspm"` property for dependencies to work. See https://github.com/jspm/registry/wiki/Configuring-Packages-for-jspm#registry.

**Comment by trusktr** [comment]
Is this only for the command line override? I think considered it safe to assume if not npm:, assume jspm:.

**Comment by trusktr** [comment]
I think we need auto here, == less friction.

**Comment by trusktr** [comment]
So infamous package.json now contains

```js
  "jspm": {
    "registry": "jspm",
    "directories": {
      "lib": "src"
    }
  },
```

and the name for infamous in the jspm registry resolves to the github repo. Looks good?

**Comment by guybedford** [comment]
Default options can be too clever. Rather use warnings to guide users. If you want to submit a PR for a warning when setting dependencies with no `registry` or `jspm` property on an override that is welcome.

**Comment by trusktr** [comment]
I think defaulting to `jspm` except for npm packages would be fine, along with documentation. The documentation would serve as the warning, plus the majority of new users will be using the npm or jspm registries, so having it not work out of the box might prevent some from adopting jspm, which we wouldn't want. The browserify and webpack workflows were way too easy, which is a good thing, and advanced users can always learn how to customize. I'd prefer sensible defaults over warnings.

The docs can say something like if installing from npm, the registry is npm. When not installing from npm, t's jspm unless you specify otherwise.

That'd make things smoother for new users I think.

**Comment by guybedford** [comment]
The idea is that overrides behave exactly the same way as the package.json itself.

A package.json with "dependencies" in it, and no "jspm" property, will not have those dependencies handled by jspm at all. So we have to be consistent with what an override means, which is not "work out what I mean and give me that override instead".

Warnings are the way to go to guide users away from issues like this, which I completely agree are massive catches.

**Comment by guybedford** [comment]
I guess the `jspm` property is like an override an automatically acts with `registry: jspm` so this is something we could automate after all.

**Comment by trusktr** [comment]
Yeah, that's what my impression was all along: that the `jspm` property *is* an override, and it didn't occur to me that I could put *jspm* in an override, which I think should just not be the case, and that the `jspm` property and overrides elsewhere are one and the same.

**Comment by trusktr** [comment]
So in the case of treating command line and registry overrides identically to the `jspm` field of a package.json, the docs could say along the lines of

> Overrides serve to override the properties found in package.json. They can be specified in three ways: as a command line argument to jspm, in a registry, and in the `jspm` property found in package.json. As expected, a `jspm` property in a package.json will override every in the package.json's root except for the `jspm` property itself.

:D

**Comment by guybedford** [comment]
Sure, PRs welcome!

**Comment by trusktr** [comment]
I figured I'd give it a shot with those two PRs I just made, to start getting more familiar with how it all works. The thing is, I come from a background with no Promise usage whatsoever, so I was quite lost in all those promise chains. I'm accustomed to good 'ol callbacks, with patterns like those in [`async`](https://www.npmjs.com/package/async) to manage flow, along with trying to modularize everything using classes.
